api/src/main/java/marquez/db/RunDao.java

```diff
@@ -80,9 +80,9 @@ public interface RunDao extends BaseDao {
           + "FROM runs_view AS r\n"
           + "LEFT OUTER JOIN\n"
           + "(\n"
-          + "    SELECT le.run_uuid, JSON_AGG(event->'run'->'facets') AS facets\n"
-          + "    FROM lineage_events le\n"
-          + "    GROUP BY le.run_uuid\n"
+          + "    SELECT rf.run_uuid, JSON_AGG(rf.facet ORDER BY rf.lineage_event_time ASC) AS facets\n"
```

Comment:
Thanks @mobuchowski for the pointing this out (which, I agree with).

@pawel-big-lebowski I know this PR / optimization has been dragging (mostly my fault, so happy to take the blame), but splitting the changes into separate PRs / releases would reduce any challenges with rollbacks, but also provide users the option to decide when to apply the migration / upgrade. So, that said, what I propose is:

1. Open a PR to just introduce the dataset, job, and run facets tables (a write only mode), meaning the facet tables won't be referenced in anyway in our queries, but will be written to so that features can take advantage of the optimization; lineage events will continue to be written to the `lineage_events` table as well. This can be released in `0.30.0`.
2. As a follow up PR, we can update all references to `lineage_events` in our queries to use the individual facet tables. Within this PR, the migration script (and documentation) to backfill facets from the `lineage_events` table will also be introduced (and be **required** to run for users with larger `lineage_events` tables). This can be released in `0.31.0`.

PR #2152 can be **step 2** and marked ias blocked, and I can cherry pick any changes into a separate PR for **step 1** (also option to other options). Thoughts?

Reply:
It may be difficult to split the change into separate PRs. The initial PR was missing tests. I added them within my commits and it detected some issues with previous commits like:
 * `runFacetExists` method in `api/src/main/java/marquez/db/RunFacetsDao.java`
 * `fromName` method in `api/src/main/java/marquez/db/DatasetFacetsDao.java` 
 (see my latest commit)

So merging some of the commits shouldn't be a solution. Additionally, one cannot take benefit from facet tables unless they are filled with data. Otherwise, users will experience incomplete results. 

This PR does not clean or remove data from `lineage_events` table so we always have the ability to prepare a fix that restores previous mechanics if we want to. 

@wslulciuc Although splitting this PR into multiple PRs could improve readability, I don't see any advantage of splitting this into separate releases (`0.30.0` and `0.31.0`).

Reply:
The combination of a manual migration process and the fact that the code stops reading from the `lineage_events` table introduces a real problem for users. When a user deploys, the tables will be created and all new records will be written into the new facet tables. However, all old data still lives only in the `lineage_events` table and won't be accessible until after the user executes the migration command. This means that the service will effectively be broken until users choose to execute this migration.

I think splitting the PR so that writes and reads are committed and can be deployed separately makes sense. This gives users the opportunity to push multiple deployments in a way that guarantees no downtime. It also makes this PR muuuch easier to review - readability is not just good for its own sake; it makes it easier to catch issues before they go to production. The migration script itself warrants its own review, I think.